src/Meilisearch/Builder/CompositeFilterBuilder.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusMeilisearchPlugin\Meilisearch\Builder;
+
+final class CompositeFilterBuilder implements FilterBuilderInterface
+{
+    /** @var array<FilterBuilderInterface> */
+    private array $filterBuilders;
+
+    public function __construct(
+        iterable $filterBuilders,
+    ) {
+        $this->filterBuilders = $filterBuilders instanceof \Traversable ? iterator_to_array($filterBuilders) : $filterBuilders;
```

Comment:
Is it even needed? \Traversable, iterable, array - all should be iterable in the same way 🖖